internal/locate/region_request.go

```diff
@@ -906,9 +906,14 @@ func newReplicaSelector(
 	regionCache *RegionCache, regionID RegionVerID, req *tikvrpc.Request, opts ...StoreSelectorOption,
 ) (*replicaSelector, error) {
 	cachedRegion := regionCache.GetCachedRegionWithRLock(regionID)
-	if cachedRegion == nil || !cachedRegion.isValid() {
-		return nil, nil
+	if cachedRegion == nil {
+		return nil, errors.New("cached region not found")
+	} else if cachedRegion.checkNeedReload() {
+		return nil, errors.New("cached region not found")
+	} else if !cachedRegion.checkRegionCacheTTL(time.Now().Unix()) {
+		return nil, errors.New("cached region not found")
```

Comment:
Seems it's ok to simply check by `isValid`.
```go
if cachedRegion.isValid() {
	return nil, errors.New("cached region not found")
}
```

Reply:
mistake error, I have fixed it.